File: buzzvil-sdk-ios-v6/buzzvil-sdk-ios-objc/BenefitHub/BenefitHubContainerViewController.m
@import BuzzvilSDK;
#import "BenefitHubContainerViewController.h"

@interface BenefitHubContainerViewController ()

@property (nonatomic, strong, readonly) BuzzBenefitHub *benefitHub;

@end

@implementation BenefitHubContainerViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  
  _benefitHub = [[BuzzBenefitHub alloc] init];
  [self displayContentViewController:[_benefitHub getViewController]];
}

- (void)displayContentViewController:(UIViewController*)contentViewController {
  [self addChildViewController:contentViewController];
  contentViewController.view.frame = self.view.bounds;
  [self.view addSubview:contentViewController.view];
  [contentViewController didMoveToParentViewController:self];
}

@end

File: buzzvil-sdk-ios-v6/buzzvil-sdk-ios-swift/BenefitHub/BenefitHubContainerViewController.swift
import BuzzvilSDK
import UIKit

// 하위 뷰 컨트롤러로 베네핏허브 연동하기
class BenefitHubContainerViewController: UIViewController {
  let benefitHub = BuzzBenefitHub()
  
  override func viewDidLoad() {
    super.viewDidLoad()
    view.backgroundColor = .systemBackground

    displayContentViewController(benefitHub.createViewController())
  }
  
  private func displayContentViewController(_ contentViewController: UIViewController) {
    addChild(contentViewController)
    contentViewController.view.frame = view.bounds
    view.addSubview(contentViewController.view)
    contentViewController.didMove(toParent: self)
  }
}
